faucet.js

```diff
@@ -16,14 +16,20 @@ class Faucet {
      * @dev constructor for faucet
      * @param url The conflux provider url 
      * @param address The faucet contract address
+     * @param lastAddress The last faucet contract address 
      */
-    constructor(url, address) {
+    constructor(url, address, lastAddress) {
         this.cfx = new Conflux({url: url});
         this.address = address;
+        this.lastAddress = lastAddress;
         this.faucet = this.cfx.Contract({
             abi: faucetContract.abi,
             address: address,
         });
+        this.oldFaucet = this.cfx.Contract({
+            abi: faucetContract.abi,
```

Comment:
Better to add necessary comment that `oldFaucet` and `newFaucet` share the same ABI file, and be careful with compatibility when use `oldFaucet`.

Reply:
ABI is compatible for this use, and a commet is added to explain this.